features/cellular/framework/targets/UBLOX/AT/UBLOX_AT_CellularStack.cpp

```diff
@@ -0,0 +1,596 @@
+/*
+ * Copyright (c) 2017, Arm Limited and affiliates.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "UBLOX_AT_CellularStack.h"
+
+using namespace mbed;
+using namespace mbed_cellular_util;
+
+UBLOX_AT_CellularStack::UBLOX_AT_CellularStack(ATHandler &atHandler, int cid, nsapi_ip_stack_t stack_type) : AT_CellularStack(atHandler, cid, stack_type)
+{
+    // URC handlers for sockets
+    _at.set_urc_handler("+UUSORD:", callback(this, &UBLOX_AT_CellularStack::UUSORD_URC));
+    _at.set_urc_handler("+UUSORF:", callback(this, &UBLOX_AT_CellularStack::UUSORF_URC));
+    _at.set_urc_handler("+UUSOCL:", callback(this, &UBLOX_AT_CellularStack::UUSOCL_URC));
+    _at.set_urc_handler("+UUPSDD:", callback(this, &UBLOX_AT_CellularStack::UUPSDD_URC));
+	
+    pendingBytes = 0;
+}
+
+UBLOX_AT_CellularStack::~UBLOX_AT_CellularStack()
+{
+}
+
+nsapi_error_t UBLOX_AT_CellularStack::socket_listen(nsapi_socket_t handle, int backlog)
+{
+    return NSAPI_ERROR_UNSUPPORTED;
+}
+
+nsapi_error_t UBLOX_AT_CellularStack::socket_accept(void *server, void **socket, SocketAddress *addr)
+{
+    return NSAPI_ERROR_UNSUPPORTED;
+}
+
+// Callback for Socket Read URC.
+void UBLOX_AT_CellularStack::UUSORD_URC()
+{
+    int a,b;
+    CellularSocket *socket;
+
+    a =_at.read_int();
+    b =_at.read_int();
+
+    socket = find_socket(a);
+    if (socket != NULL) {
+        socket->rx_avail = true;
+        pendingBytes = b;
```

Comment:
Doesn't the `pendingBytes` value need to be stored per socket (otherwise it will be confused if we have more than one socket running at the same time)?

Reply:
My apologies. It was in my mind, forgot to discuss. Previously SockCtrl structure have pendingByte attribute but now CellularSocket class don't have any attribute for pendingBytes. So how to handle it.

Reply:
As discussed, we could (as a first step) add a `pendingBytes` field to the `CellularSockets` data structure.  If ARM object to this when we submit the PR then we can ask them for their solution.